pkg/yaml/yaml.go

```diff
@@ -24,15 +26,62 @@ var (
 	cleanContains = []string{
 		"cattle.io/",
 	}
+	buffSize = 4096
 )
 
+// Unmarshal decodes YAML bytes into document (as defined by
+// the YAML spec) then converting it to JSON via
+// "k8s.io/apimachinery/pkg/util/yaml".YAMLToJSON and then decoding the json in the the v interface{}
 func Unmarshal(data []byte, v interface{}) error {
```

Comment:
Is this function similar to the new one? We just shouldn't use it anymore because it's less safe the new, generic one?

Reply:
The main difference is the new one works with YAML, which contains a list of docs or a single doc. `Unmarshal(...)` only works for bytes that contain a single document. The other function that already exists to help with lists is `ToObject(...)`, but this function forces you to use an `unstructured.Unstructured` objects which as you know cause a lot of headaches when you try to convert them to actual objects with map decode.

`Unmarshal(...)` is still safe if you know your YAML file is not a list of objects. `UnmarshalWithJSONDecoder(...)` provides more functionality, I don't want to depreciate the function since if you are using it today there is no reason to change.